doc/fundamentals/updating_pipelines.md

```diff
@@ -1,41 +1,63 @@
+In cases in which you are updating parallelism, adding another input repo, or
+otherwise modifying your [pipeline
+specification](../reference/pipeline_spec.html), you just need to update your
+JSON file and call `update-pipeline`:
 
-```sh	
-$ pachctl update-pipeline -f pipeline.json 
+```sh
+$ pachctl update-pipeline -f pipeline.json
 ```
 
-Similar to `create-pipeline`, `update-pipeline` with the `-f` flag can also take a URL if your JSON manifest is hosted on GitHub or elsewhere. 
+Similar to `create-pipeline`, `update-pipeline` with the `-f` flag can also
+take a URL if your JSON manifest is hosted on GitHub or elsewhere.
 
 ## Updating the code used in a pipeline
 
-You can also use `update-pipeline` to update the code you are using in one or more of your piplines.  To update the code in your pipeline:
+You can also use `update-pipeline` to update the code you are using in one or
+more of your piplines.  To update the code in your pipeline:
 
 1. Make the code changes.
 2. Re-build your Docker image.
 3. Call `update-pipeline` with the `--push-images` flag.
 
-You need to call `update-pipeline` with the `--push-images` flag because, if you have already run your pipeline, Pachyderm has already pulled the specified images.  It won't re-pull new versions of the images, unless we tell it to (which ensures that we don't waste time pulling images when we don't need to).  When `--push-images` is specified, Pachyderm will do the following:
+You need to call `update-pipeline` with the `--push-images` flag because, if
+you have already run your pipeline, Pachyderm has already pulled the specified
+images.  It won't re-pull new versions of the images, unless we tell it to
+(which ensures that we don't waste time pulling images when we don't need to).
+When `--push-images` is specified, Pachyderm will do the following:
 
 1. Tag your image with a new unique tag.
 2. Push that tagged image to your registry (e.g., DockerHub).
-3. Update the pipeline specification that you previously gave to Pachyderm with the new unique tag.
+3. Update the pipeline specification that you previously gave to Pachyderm with
+   the new unique tag.
 
-For example, you could update the Python code used in the [OpenCV pipeline](../getting_started/beginner_tutorial.html) via:
+For example, you could update the Python code used in the [OpenCV
+pipeline](../getting_started/beginner_tutorial.html) via:
 
 ```sh
 pachctl update-pipeline -f edges.json --push-images --password <registry password> -u <registry user>
 ```
 
-## Re-processing commits, `from` commit
-
-Changing your pipeline code implies that your previously computed results aren't in sync with (or generated by) your most recent code.  By default (if the "from-commit" field in the pipeline spec is not given), Pachyderm will start a new "commit tree" for your new code and re-compute the results with your new code (committing to the new commit tree). 
+## Re-processing data
```

Comment:
This is the only section that actually changed, the other stuff is just formatting.